tests/queries/tests.py

```diff
@@ -2248,6 +2248,44 @@ def test_field_error_values_list(self):
         with self.assertRaisesMessage(FieldError, msg):
             Tag.objects.values_list('name__foo')
 
+    def test_named_values_list_flat(self):
+        msg = "'flat' and 'named' can't be passed together."
+        with self.assertRaisesMessage(TypeError, msg):
+            Number.objects.values_list('num', flat=True, named=True)
+
+    def test_named_values_list_bad_field_name(self):
+        msg = "Type names and field names must be valid identifiers: '1'"
```

Comment:
Did you consider catching this exception and reraising a message that explains the issue in terms that a Django user might more easily understand?

Reply:
How about this?
```
try:
    tuple_class = self.create_namedtuple_class(*names)
except Exception as e:
    msg = 'namedtuple() raised exception. Make sure that field names are valid identifiers.'
    raise ValueError(msg) from e
```

Reply:
If only `QuerySet.extra()` is affected, we can skip it since `extra()` usage is discouraged.

Reply:
I concur